pandas/tests/frame/methods/test_diff.py

```diff
@@ -74,7 +74,7 @@ def test_diff_datetime_axis1(self, tz):
             )
             tm.assert_frame_equal(result, expected)
         else:
-            with pytest.raises(NotImplementedError):
+            with pytest.raises(NotImplementedError, match=""):
```

Comment:
Hmm what does this actually do? Isn't this the same as not providing a match argument?

Reply:
This checks that match argument is an empty string. If I provide any other msg (not msg = ""), it throws an error. Should I remove match=""?

Reply:
What is the error? Can you not add the expected pattern to match against?

Reply:
What I meant is that a matching argument is an empty string. For example, if I put
            msg = "my error message"
            with pytest.raises(NotImplementedError, match = msg):
Then I get the error AssertionError: Pattern 'my error message' not found in ''. So I put a matching argument match = "". Though if I shouldn't match against an empty string, I can remove it and keep it as it was before:
            with pytest.raises(NotImplementedError):

Reply:
Sure go ahead and revert; I don't know what to do on this one specifically so can tackle later. The rest of the changes look good

Reply:
Sure will do, thanks